packages/edit-site/src/components/global-styles/screen-revisions/get-revision-changes.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+const globalStylesChangesCache = new WeakMap();
+
+const translationMap = {
+	caption: __( 'caption' ),
+	link: __( 'link' ),
+	button: __( 'button' ),
+	heading: __( 'heading' ),
+	'settings.color': __( 'color settings' ),
+	'settings.typography': __( 'typography settings' ),
+	'color.text': __( 'text colors' ),
+	'color.background': __( 'background colors' ),
+	'spacing.margin': __( 'margin styles' ),
+	'spacing.padding': __( 'padding styles' ),
+	'spacing.blockGap': __( 'block spacing' ),
+	'settings.layout': __( 'layout settings' ),
+	'typography.fontStyle': __( 'font style' ),
+	'typography.fontSize': __( 'font size' ),
+	'typography.lineHeight': __( 'line height' ),
+	'typography.fontFamily': __( 'font family' ),
+	'typography.fontWeight': __( 'font weight' ),
+};
+
+const isObject = ( obj ) => obj !== null && typeof obj === 'object';
+
+/**
+ * Get the translation for a given global styles key.
+ * @param {string}                key        A key representing a path to a global style property or setting.
+ * @param {Record<string,string>} blockNames A key/value pair object of block names and their rendered titles.
+ * @return {string|undefined}                A translated key or undefined if no translation exists.
+ */
+function getTranslation( key, blockNames ) {
+	if ( translationMap[ key ] ) {
+		return translationMap[ key ];
+	}
+
+	const keyArray = key.split( '.' );
+
+	if ( keyArray?.[ 0 ] === 'blocks' ) {
+		const blockName = blockNames[ keyArray[ 1 ] ];
+		return blockName
+			? sprintf(
+					// translators: %s: block name.
+					__( '%s block' ),
+					blockName
+			  )
+			: keyArray[ 1 ];
+	}
+
+	if ( keyArray?.[ 0 ] === 'elements' ) {
+		return sprintf(
+			// translators: %s: element name, e.g., heading button, link, caption.
+			__( '%s element' ),
+			translationMap[ keyArray[ 1 ] ]
+		);
+	}
+
+	return undefined;
+}
+
+/**
+ * A deep comparison of two objects, optimized for comparing global styles.
+ * @param {Object} changedObject  The changed object to compare.
+ * @param {Object} originalObject The original object to compare against.
+ * @param {string} parentPath     A key/value pair object of block names and their rendered titles.
+ * @return {string[]}             An array of paths whose values have changed.
+ */
+function deepCompare( changedObject, originalObject, parentPath = '' ) {
+	// We have two non-object values to compare.
+	if ( ! isObject( changedObject ) && ! isObject( originalObject ) ) {
+		// Only return a path if the value has changed.
+		// And then only the path name up to 2 levels deep.
+		return changedObject !== originalObject
```

Comment:
So this is very specific to the UI's requirements.

We're just returning the path to a value that has changed between versions.

If we want to reuse this function later, we can return anything, e.g., an object:

```js
{
    path: 'styles.typography.fontWeight',
    oldValue: 600,
    newValue: 300,
}
```